addons/product/models/product_template.py

```diff
@@ -406,14 +406,23 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
             return super(ProductTemplate, self)._name_search(name=name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid)
 
         Product = self.env['product.product']
+        Template = self.env['product.template']
         templates = self.browse([])
+        domain_no_variant = [('product_variant_ids', '=', False)]
+        domain_no_variant = domain_no_variant + args if args else domain_no_variant
```

Comment:
useless

Reply:
which part ? the ternary ? args can be None
The condition ? maybe